src/visitors/implicit_argument_visitor.cpp

```diff
@@ -0,0 +1,36 @@
+/*************************************************************************
+ * Copyright (C) 2022 Blue Brain Project
+ *
+ * This file is part of NMODL distributed under the terms of the GNU
+ * Lesser General Public License. See top-level LICENSE file for details.
+ *************************************************************************/
+#include "visitors/implicit_argument_visitor.hpp"
+#include "ast/function_call.hpp"
+#include "ast/string.hpp"
+
+#include <cassert>
+#include <iostream>
+
+namespace nmodl {
+namespace visitor {
+
+// TODO should this also handle the case covered by
+// needs_neuron_thread_first_arg?
```

Comment:
Looking at `extern_definitions`, nothing comes to mind that needs to be changed (immediately).

Reply:
I think what I meant here was that we now have two different ways of fudging extra arguments into function calls:
- this visitor, which I use for `nrn_ghk` and which actually modifies the AST
- this code in the code printing: https://github.com/BlueBrain/nmodl/blob/40f046c7aaf22af7e265efb52ea68f37195c4516/src/codegen/codegen_c_visitor.cpp#L1382-L1388

and I was wondering if we should consolidate and only use one approach. Originally I think I tried with the extra visitor because I was trying to only add `celsius` to the instance struct if it was actually used, so adding that information to the AST was useful.

Reply:
Done.